CalibCalorimetry/EcalLaserSorting/src/WatcherStreamFileReader.cc

```diff
@@ -392,7 +392,9 @@ edm::StreamerInputFile* WatcherStreamFileReader::getInputFile() {
   return streamerInputFile_.get();
 }
 
-void WatcherStreamFileReader::closeFile() {
+void WatcherStreamFileReader::closeFile() {}
+
+void WatcherStreamFileReader::moveJustReadFile() {
```

Comment:
For my understanding:
- what's the advantage of renaming this method?
- should the `closeFile()` method still remain, then?

Reply:
I believe `closeFile` is called by code that calls this class. As the semantics of the function did not match the expected behavior of the code calling `closeFile` I made a new function which explicitly states what is happening.

Reply:
> I believe `closeFile` is called by code that calls this class. As the semantics of the function did not match the expected behavior of the code calling `closeFile` I made a new function which explicitly states what is happening.

And what happens if the code that (possibly) still calls `closeFile()` finds an empty methond, instead of the previous one?

Reply:
> And what happens if the code that (possibly) still calls closeFile() finds an empty methond, instead of the previous one?

Well, closeFile should be called when the framework tells the Source that it should close the file. It happens that WatcherStreamFileReader violates the behavior of the framework by closing and opening files without telling the framework. As this change required me to add an 'artificial' file close (i.e. tell the framework to do a `File` transition even though we are not necessarily on a file boundary) the fileClose method will be called at a time where this module does not want to move the read file. That meant I had to move the functionality of `moveJustReadFile` to its own function. In addition, I didn't want to make WatcherStreamFileReader actually behave as the framework expected (i.e. informing the framework that new file transitions are needed) as I was unsure how that would affect the performance of jobs using that Source.